--- calibre/setup/installer/linux/freeze2.py	2010-05-07 20:32:02.000000000 +0200
+++ calibre/setup/installer/linux/freeze2.py	2010-05-10 22:12:01.676321791 +0200
@@ -71,7 +71,7 @@ class LinuxFreeze2(Command):
         #self.initbase()
         #self.copy_libs()
         #self.copy_python()
-        #self.compile_mount_helper()
+        self.compile_mount_helper()
         self.build_launchers()
 
     def initbase(self):
