--- calibre/src/calibre/utils/localization.py~	2010-05-21 11:47:23.959845665 +0200
+++ calibre/src/calibre/utils/localization.py	2010-05-21 11:47:07.293172188 +0200
@@ -45,7 +45,7 @@
     return lang
 
 def messages_path(lang):
-    return P('localization/locales/%s/LC_MESSAGES'%lang)
+    return ('/usr/share/locale/%s/LC_MESSAGES'%lang)
 
 def get_lc_messages_path(lang):
     hlang = None
@@ -75,9 +75,9 @@
         mpath = get_lc_messages_path(lang)
         if mpath is not None:
             if buf is None:
-                buf = open(os.path.join(mpath, 'messages.mo'), 'rb')
+                buf = open(os.path.join(mpath, 'calibre.mo'), 'rb')
             mpath = mpath.replace(os.sep+'nds'+os.sep, os.sep+'de'+os.sep)
-            isof = os.path.join(mpath, 'iso639.mo')
+            isof = os.path.join(mpath, 'calibre_iso639.mo')
             if os.path.exists(isof):
                 iso639 = open(isof, 'rb')
 
--- calibre/src/calibre/startup.py~	2010-06-06 22:46:25.000000000 +0200
+++ calibre/src/calibre/startup.py	2010-06-11 20:31:39.644201974 +0200
@@ -49,7 +49,7 @@
     # Setup translations
     from calibre.utils.localization import set_translators
 
-    set_translators()
+    #set_translators()
 
     ################################################################################
     # Initialize locale
