diff -urNp -x '*.orig' calibre-4.20.0.org/setup/install.py calibre-4.20.0/setup/install.py
--- calibre-4.20.0.org/setup/install.py	2020-07-03 04:11:13.000000000 +0200
+++ calibre-4.20.0/setup/install.py	2021-06-23 14:08:38.902233250 +0200
@@ -100,15 +100,10 @@ class Develop(Command):
             self.sharedir = self.j(opts.prefix, 'share')
         if not getattr(opts, 'staging_root', None):
             opts.staging_root = opts.prefix
-        self.staging_libdir = getattr(opts, 'staging_libdir', None)
-        if self.staging_libdir is None:
-            self.staging_libdir = opts.staging_libdir = self.j(opts.staging_root, 'lib')
-        self.staging_bindir = getattr(opts, 'staging_bindir', None)
-        if self.staging_bindir is None:
-            self.staging_bindir = opts.staging_bindir = self.j(opts.staging_root, 'bin')
-        self.staging_sharedir = getattr(opts, 'staging_sharedir', None)
-        if self.staging_sharedir is None:
-            self.staging_sharedir = opts.staging_sharedir = self.j(opts.staging_root, 'share')
+
+        self.staging_libdir = opts.staging_libdir = opts.staging_root + self.libdir
+        self.staging_bindir = opts.staging_bindir = self.j(opts.staging_root, 'usr/', 'bin')
+        self.staging_sharedir = opts.staging_sharedir = self.j(opts.staging_root, 'usr/', 'share')
 
         self.staging_libdir = opts.staging_libdir = self.j(self.staging_libdir, 'calibre')
         self.staging_sharedir = opts.staging_sharedir = self.j(self.staging_sharedir, 'calibre')
diff -urNp -x '*.orig' calibre-4.20.0.org/src/calibre/linux.py calibre-4.20.0/src/calibre/linux.py
--- calibre-4.20.0.org/src/calibre/linux.py	2020-07-03 04:11:13.000000000 +0200
+++ calibre-4.20.0/src/calibre/linux.py	2021-06-23 14:08:38.902233250 +0200
@@ -737,10 +737,10 @@ class PostInstall:
             self.warn = self.warning
 
         if not self.opts.staging_bindir:
-            self.opts.staging_bindir = os.path.join(self.opts.staging_root,
+            self.opts.staging_bindir = os.path.join(self.opts.staging_root, 'usr',
             'bin')
         if not self.opts.staging_sharedir:
-            self.opts.staging_sharedir = os.path.join(self.opts.staging_root,
+            self.opts.staging_sharedir = os.path.join(self.opts.staging_root, 'usr',
             'share', 'calibre')
         self.opts.staging_etc = '/etc' if self.opts.staging_root == '/usr' else \
                 os.path.join(self.opts.staging_root, 'etc')
@@ -748,7 +748,7 @@ class PostInstall:
         prefix = getattr(self.opts, 'prefix', None)
         if prefix and prefix != self.opts.staging_root:
             self.opts.staged_install = True
-            os.environ['XDG_DATA_DIRS'] = os.path.join(self.opts.staging_root, 'share')
+            os.environ['XDG_DATA_DIRS'] = os.path.join(self.opts.staging_root, 'usr', 'share')
             os.environ['XDG_UTILS_INSTALL_MODE'] = 'system'
 
         from calibre.utils.serialize import msgpack_loads
