--- calibre/src/calibre/gui2/update.py~	2010-06-20 02:32:33.000000000 +0200
+++ calibre/src/calibre/gui2/update.py	2010-06-23 12:36:15.515849515 +0200
@@ -40,11 +40,11 @@
 class UpdateMixin(object):
 
     def __init__(self, opts):
-        if not opts.no_update_check:
-            self.update_checker = CheckForUpdates(self)
-            self.update_checker.update_found.connect(self.update_found,
-                    type=Qt.QueuedConnection)
-            self.update_checker.start()
+        #if not opts.no_update_check:
+        #    self.update_checker = CheckForUpdates(self)
+        #    self.update_checker.update_found.connect(self.update_found,
+        #            type=Qt.QueuedConnection)
+        #    self.update_checker.start()
 
     def update_found(self, version):
         os = 'windows' if iswindows else 'osx' if isosx else 'linux'
