--- calibre/src/calibre/gui2/ui.py~	2010-06-06 22:46:26.000000000 +0200
+++ calibre/src/calibre/gui2/ui.py	2010-06-11 19:52:30.581117740 +0200
@@ -266,11 +266,11 @@
         self.latest_version = ' '
         self.vanity.setText(self.vanity_template%dict(version=' ', device=' '))
         self.device_info = ' '
-        if not opts.no_update_check:
-            self.update_checker = CheckForUpdates(self)
-            self.update_checker.update_found.connect(self.update_found,
-                    type=Qt.QueuedConnection)
-            self.update_checker.start()
+        #if not opts.no_update_check:
+        #    self.update_checker = CheckForUpdates(self)
+        #    self.update_checker.update_found.connect(self.update_found,
+        #            type=Qt.QueuedConnection)
+        #    self.update_checker.start()
         ####################### Status Bar #####################
         self.status_bar.initialize(self.system_tray_icon)
         self.status_bar.show_book_info.connect(self.show_book_info)
