--- calibre/src/calibre/gui2/ui.py~	2010-07-09 23:38:12.000000000 +0200
+++ calibre/src/calibre/gui2/ui.py	2010-07-14 13:38:25.919379342 +0200
@@ -204,7 +204,7 @@
         self.eject_action.triggered.connect(self.device_manager.umount_device)
 
         #################### Update notification ###################
-        UpdateMixin.__init__(self, opts)
+        #UpdateMixin.__init__(self, opts)
 
         ####################### Setup Toolbar #####################
         ToolbarMixin.__init__(self)
