--- calibre/recipes/nrc-nl-epub.recipe~	2011-03-26 02:30:09.000000000 +0100
+++ calibre/recipes/nrc-nl-epub.recipe	2011-03-29 09:31:15.404987337 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env  python2
+#!/usr/bin/env  python
 # -*- coding: utf-8 -*-
 #Based on veezh's original recipe and Kovid Goyal's New York Times recipe
 
--- calibre/setup/install.py~	2011-03-29 09:28:50.238365816 +0200
+++ calibre/setup/install.py	2011-03-29 09:32:00.301640042 +0200
@@ -12,7 +12,7 @@
         __appname__, __version__
 
 HEADER = '''\
-#!/usr/bin/env python2
+#!/usr/bin/env python
 
 """
 This is the standard runscript for all of calibre's tools.
