diff -urNp -x '*.orig' calibre-4.20.0.org/setup/install.py calibre-4.20.0/setup/install.py
--- calibre-4.20.0.org/setup/install.py	2021-06-23 14:08:46.395856286 +0200
+++ calibre-4.20.0/setup/install.py	2021-06-23 14:08:48.429265433 +0200
@@ -20,7 +20,7 @@ from setup import (
 )
 
 HEADER = '''\
-#!/usr/bin/env python{py_major_version}
+#!/usr/bin/env python
 
 """
 This is the standard runscript for all of calibre's tools.
